go/host/storage/storage.go

```diff
@@ -42,12 +42,15 @@ func (s *storageImpl) AddBatch(batch *common.ExtBatch) error {
 			return err1
 		}
 		if errors.Is(err, errutil.ErrAlreadyExists) {
-			return err
+			return nil
 		}
 		return fmt.Errorf("could not add batch to host. Cause: %w", err)
 	}
 
 	if err := dbtx.Write(); err != nil {
+		if errors.Is(err, errutil.ErrAlreadyExists) {
```

Comment:
I think this has to check over the text. The error here will be returned directly from the db